library/common/extensions/filters/http/assertion/filter.cc

```diff
@@ -0,0 +1,79 @@
+#include "library/common/extensions/filters/http/assertion/filter.h"
+
+#include "envoy/http/codes.h"
+#include "envoy/server/filter_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Assertion {
+
+AssertionFilterConfig::AssertionFilterConfig(
+    const envoymobile::extensions::filters::http::assertion::Assertion& proto_config) {
+  Common::Tap::buildMatcher(proto_config.match_config(), matchers_);
+}
+
+Extensions::Common::Tap::Matcher& AssertionFilterConfig::rootMatcher() const {
+  ASSERT(!matchers_.empty());
+  return *matchers_[0];
+}
+
+AssertionFilter::AssertionFilter(AssertionFilterConfigSharedPtr config) : config_(config) {
```

Comment:
Can we add a docstring to this? It would also be really helpful to link the matcher documentation (whatever there is).

Reply:
Already exists in the header.

Reply:
Yep, you're right. Can you still add a link to documentation on matchers?

Reply:
The proto is linked. I think the larger issue is that we need to start generating docs from our protos, cut this issue to track: https://github.com/lyft/envoy-mobile/issues/1001